internal/testrunner/runners/system/runner.go

```diff
@@ -69,36 +69,51 @@ func Run(options testrunner.TestOptions) error {
 	return r.run()
 }
 
-func (r *runner) run() error {
+func (r *runner) run() ([]testrunner.TestResult, error) {
+	result := testrunner.TestResult{
+		Package:    r.testFolder.Package,
+		DataStream: r.testFolder.DataStream,
+	}
+	startTime := time.Now()
+	resultsWith := func(tr testrunner.TestResult, err error) ([]testrunner.TestResult, error) {
```

Comment:
I understand your shortcut function, but I admit I had to figure out why is it called "with" :) At the moment the system test runner supports only single test case per data stream, but it's possible that we'll add more test cases in the future. It would be great to make the test reporter ready for this.

Reply:
I wonder if we need a test results builder with fluent API for such cases.

Reply:
Thanks for the feedback on the name. I will find a better name for this function.

As for the suggestions to prepare for multiple test cases and providing a fluent API, as always I would prefer to defer these changes until we need them. Generally speaking, I'm a fan of YAGNI but I'm not super religious about it either so if you feel strongly about either of these suggestions I can go ahead and implement them now itself 🙂 .

Reply:
YAGNI... hmm... ok, you're right ;) so far, testResult and error go together, let's leave it as a function.